stdlib/2/_io.pyi

```diff
@@ -19,91 +19,149 @@ class _IOBase(BinaryIO):
+    def __init__(self, reader: _RawIOBase, writer: _RawIOBase) -> None: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def __enter__(self) -> 'BufferedRWPair': ...
 
 class BufferedRandom(_BufferedIOBase):
+    mode = ...  # type: str
     name = ...  # type: str
     raw = ...  # type: _IOBase
-    mode = ...  # type: str
-    def peek(self, n: int = ...) -> str: ...
+    def __init__(self, raw: _IOBase,
+                 buffer_size: int = ...,
+                 max_buffer_size: int = ...) -> None: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def __enter__(self) -> 'BufferedRandom': ...
 
 class BufferedReader(_BufferedIOBase):
+    mode = ...  # type: str
     name = ...  # type: str
     raw = ...  # type: _IOBase
-    mode = ...  # type: str
-    def peek(self, n: int = ...) -> str: ...
+    def __init__(self, raw: _IOBase, buffer_size: int = ...) -> None: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def __enter__(self) -> 'BufferedReader': ...
 
 class BufferedWriter(_BufferedIOBase):
     name = ...  # type: str
     raw = ...  # type: _IOBase
     mode = ...  # type: str
+    def __init__(self, raw: _IOBase,
+                 buffer_size: int = ...,
+                 max_buffer_size: int = ...) -> None: ...
+    def __enter__(self) -> 'BufferedWriter': ...
 
 class BytesIO(_BufferedIOBase):
+    def __init__(self, initial_bytes: bytes = ...) -> None: ...
     def __setstate__(self, tuple) -> None: ...
     def __getstate__(self) -> tuple: ...
-    def getvalue(self) -> str: ...
+    def getvalue(self) -> bytes: ...
+    def write(self, s: bytes) -> int: ...
+    def writelines(self, lines: Iterable[bytes]) -> None: ...
+    def read1(self, size: int) -> bytes: ...
+    def __iter__(self) -> Iterator[bytes]: ...
+    def next(self) -> bytes: ...
+    def __enter__(self) -> 'BytesIO': ...
 
 class _RawIOBase(_IOBase):
     def readall(self) -> str: ...
-    def read(self, n: int = ...) -> str: ...
+    def read(self, size: int = ...) -> str: ...
+    def __enter__(self) -> '_RawIOBase': ...
 
-class FileIO(_RawIOBase):
+class FileIO(_RawIOBase, BytesIO):  # type: ignore  # for __enter__
     mode = ...  # type: str
     closefd = ...  # type: bool
+    def __init__(self, file: str, mode: str = ...) -> None: ...
     def readinto(self, buffer: bytearray)-> int: ...
     def write(self, pbuf: str) -> int: ...
+    def __enter__(self) -> 'FileIO': ...
 
 class IncrementalNewlineDecoder(object):
     newlines = ...  # type: Union[str, unicode]
+    def __init__(self, decoder, translate, z=...) -> None: ...
     def decode(self, input, final) -> Any: ...
     def getstate(self) -> Tuple[Any, int]: ...
     def setstate(self, state: Tuple[Any, int]) -> None: ...
     def reset(self) -> None: ...
 
-class _TextIOBase(_IOBase):
+
+# Note: In the actual _io.py, _TextIOBase inherits from _IOBase.
+class _TextIOBase(TextIO):
     errors = ...  # type: Optional[str]
     newlines = ...  # type: Union[str, unicode]
```

Comment:
Actually, on `_TextIOBase`, this always returns None:

https://github.com/python/cpython/blob/2.7/Modules/_io/textio.c#L104

overloaded below.